src/main/java/com/tudu/todoapp/rest/controllers/ToDoItemController.java

```diff
@@ -0,0 +1,85 @@
+package com.tudu.todoapp.rest.controllers;
+
+import com.tudu.todoapp.entities.TodoItem;
+import com.tudu.todoapp.repositories.TodoItemRepository;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1/todoitems")
+public class ToDoItemController {
+
+    private final TodoItemRepository todoItemRepository;
+
+    @Autowired
```

Comment:
Use @RequiredArgsConstructor instead of writing the constructor as you will have to change it if you want to add more autowired fields.